http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -1596,6 +1699,59 @@ class GKEInterpreter[F[_]](
     ) ++ configs ++ galaxyRestoreSettings
   }
 
+  private[util] def buildRStudioAppChartOverrideValuesString(
+    appName: AppName,
+    cluster: KubernetesCluster,
+    nodepoolName: NodepoolName,
+    namespaceName: NamespaceName,
+    disk: PersistentDisk,
+    ksaName: ServiceAccountName,
+    userEmail: WorkbenchEmail,
+    stagingBucket: GcsBucketName
+  ): List[String] = {
+    val proxyPath = s"/proxy/google/v1/apps/${cluster.cloudContext.asString}/${appName.value}/rstudio-service"
+    val k8sProxyHost = kubernetesProxyHost(cluster, config.proxyConfig.proxyDomain).address
+    val leoProxyhost = config.proxyConfig.getProxyServerHostName
+
+    val rewriteTarget = "$2"
+    val ingress = List(
+      raw"""ingress.enabled=true""",
+      raw"""ingress.annotations.nginx\.ingress\.kubernetes\.io/auth-tls-pass-certificate-to-upstream=true""",
+      raw"""ingress.annotations.nginx\.ingress\.kubernetes\.io/auth-tls-secret=${namespaceName.value}/ca-secret""",
+      raw"""ingress.annotations.nginx\.ingress\.kubernetes\.io/auth-tls-verify-client=on""",
+      raw"""ingress.annotations.nginx\.ingress\.kubernetes\.io/auth-tls-verify-depth=1""",
+      raw"""ingress.annotations.nginx\.ingress\.kubernetes\.io/ssl-redirect=true""",
+      raw"""ingress.annotations.nginx\.ingress\.kubernetes\.io/proxy-redirect-from=https://${k8sProxyHost}""",
+      raw"""ingress.annotations.nginx\.ingress\.kubernetes\.io/proxy-redirect-to=${leoProxyhost}""",
+      raw"""ingress.annotations.nginx\.ingress\.kubernetes\.io/rewrite-target=/${rewriteTarget}""",
+      raw"""ingress.hosts[0].host=${k8sProxyHost}""",
+      raw"""ingress.hosts[0].paths[0]=${proxyPath}${"(/|$)(.*)"}""",
+      raw"""ingress.tls[0].secretName=tls-secret""",
+      raw"""ingress.tls[0].hosts[0]=${k8sProxyHost}"""
+    )
+
+    val welder = List(
+      raw"""welder.extraEnv.name=GOOGLE_PROJECT""",
+      raw"""welder.extraEnv.value=${cluster.cloudContext.asString}""",
+      raw"""welder.extraEnv.name=STAGING_BUCKET""",
+      raw"""welder.extraEnv.value=${stagingBucket.value}""",
+      raw"""welder.extraEnv.name=CLUSTER_NAME""",
+      raw"""welder.extraEnv.value=${cluster.clusterName.value}""",
+      raw"""welder.extraEnv.name=OWNER_EMAIL""",
+      raw"""welder.extraEnv.value=${userEmail.value}"""
+    )
+
+    List(
+      // Node selector
+      raw"""nodeSelector.cloud\.google\.com/gke-nodepool=${nodepoolName.value}""",
+      // Persistence
+      raw"""persistence.size=${disk.size.gb.toString}G""",
+      raw"""persistence.gcePersistentDisk=${disk.name.value}""",
```

Comment:
OK so we are attaching the persistent disk to the app here. We may want to introduce a `FormattedBy.RStudio` enum to the disk object. I'm also curious what happens at app deletion time -- can it successfully detach the disk and re-attach to a new app instance? I know Galaxy handles that today but not honestly sure how much of the disk logic is portable to other apps.

Reply:
Yeah I know for Galaxy we explicitly tried to support detachable-PDs. I don't think Cromwell actually uses a PD for anything (it doesn't store any state on the disk). For RStudio though we probably actually want to support detachable PDs if possible.

Reply:
Soooo, I was hoping to avoid having to pass custom environment variables for this app but it looks like the PD reattachement logic depends on knowing the `WORKSPACE_NAME_KEY` so I need to make a small follow up commit tomorrow to add that as part of the app install.

Reply:
`WORKSPACE_NAME_KEY ` or `WORKSPACE_NAME`? Cromwell requires `WORKSPACE_NAME` and I just merge a PR in AoU to set that for both RStudio and Cromwell. If RStudio just needs WORKSPACE_NAME then it would work for now.

Reply:
They should be the same: `val WORKSPACE_NAME_KEY = "WORKSPACE_NAME"`,  Looking into the db now to understand how this information is stored and maybe it is just a matter of sending a create app request with the `WORKSPACE_NAME` in it as a custom environment variable, maybe there is no code change needed

Reply:
Yep, everything works great and the disk is reattaching without any lost data! Just need to send the createApp request with a `WORKSPACE_NAME`, I will update the testing steps above :)

Reply:
🎉